javalib/src/main/scala/java/lang/String.scala

```diff
@@ -1,42 +1,890 @@
+              } else {
+                i += 1
+              }
+            }
+
+            true
+          }
+        case _ =>
+          false
+      }
+    }
+  }
+
+  def equalsIgnoreCase(string: _String): scala.Boolean = {
+    if (string == this) {
+      true
+    } else if (string == null || count != string.count) {
+      false
+    } else {
+      var o1 = offset
+      var o2 = string.offset
+      while (o1 < offset + count) {
+        o1 += 1
+        o2 += 1
+        val c1 = value(o1)
+        val c2 = string.value(o2)
+        if (c1 != c2 && toUpperCase(c1) != toUpperCase(c2) &&
+            toLowerCase(c1) != toLowerCase(c2)) {
+          return false
+        }
+      }
+      true
+    }
+  }
+
+  def getBytes(): Array[scala.Byte] = {
+    val buffer = defaultCharset().encode(CharBuffer.wrap(value, offset, count))
+    val bytes  = new Array[scala.Byte](buffer.limit())
+    buffer.get(bytes)
+    bytes
+  }
+
+  @Deprecated
+  def getBytes(
+      start: Int, _end: Int, data: Array[scala.Byte], _index: Int): Unit = {
+    var end   = _end
+    var index = _index
+    if (0 <= start && start <= end && end <= count) {
+      end += offset
+
+      try {
+        var i = offset + start
+        while (i < end) {
+          index += 1
+          data(index) = value(i).toByte
+          i += 1
+        }
+      } catch {
+        case e: ArrayIndexOutOfBoundsException =>
+          throw new StringIndexOutOfBoundsException()
+      }
+    } else {
+      throw new StringIndexOutOfBoundsException()
+    }
+  }
+
+  def getBytes(encoding: _String): Array[scala.Byte] = {
+    val buffer =
+      getCharset(encoding).encode(CharBuffer.wrap(value, offset, count))
+    val bytes = new Array[scala.Byte](buffer.limit())
+    buffer.get(bytes)
+    bytes
+  }
+
+  private def getCharset(encoding: _String): Charset = {
+    var charset = _String.lastCharset
+
+    if (charset == null || !encoding.equalsIgnoreCase(charset.name())) {
+      try {
+        charset = Charset.forName(encoding)
+      } catch {
+        case e: IllegalCharsetNameException =>
+          throw new UnsupportedEncodingException(encoding).initCause(e)
+        case e: UnsupportedCharsetException =>
+          throw new UnsupportedEncodingException(encoding).initCause(e)
+      }
+
+      _String.lastCharset = charset
+    }
+    charset
+  }
+
+  def getBytes(encoding: Charset): Array[scala.Byte] = {
+    val buffer = encoding.encode(CharBuffer.wrap(value, offset, count))
+    val bytes  = new Array[scala.Byte](buffer.limit())
+    buffer.get(bytes)
+    bytes
+  }
+
+  def getChars(start: Int, end: Int, buffer: Array[Char], index: Int): Unit = {
+    if (0 <= start && start <= end && end <= count) {
+      System.arraycopy(value, start + offset, buffer, index, end - start)
+    } else {
+      throw new StringIndexOutOfBoundsException()
+    }
+  }
+
+  override def hashCode(): Int = {
+    if (cashedHashCode == 0) {
+      if (count == 0) {
+        return 0
```

Comment:
The `count == 0` part is unnecessary. The loop below will correctly compute `0` if `count == 0`. There's no need for that special-case.